jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -1,13 +1,309 @@
+        self.component_models = [x for x in self.component_models
+                                 if x["id"] != event]
+        del(self._initialized_models[event])
+
+    def vue_save_model(self, event):
+        if self.model_save_path[-1] == "/":
+            connector = ""
+        else:
+            connector = "/"
+        full_path = self.model_save_path + connector + self.model_label + ".pkl"
+        with open(full_path, 'wb') as f:
+            pickle.dump(self._fitted_model, f)
+
+    def vue_equation_changed(self, event):
+        # Length is a dummy check to test the infrastructure
+        if len(self.model_equation) > 20:
+            self.eq_error = True
+
+    def vue_model_fitting(self, *args, **kwargs):
+        """
+        Run fitting on the initialized models, fixing any parameters marked
+        as such by the user, then update the displauyed parameters with fit
+        values
+        """
+        fitted_model, fitted_spectrum = fit_model_to_spectrum(
+            self._spectrum1d,
+            self._initialized_models.values(),
+            self.model_equation,
+            run_fitter=True)
+        self._fitted_model = fitted_model
+        self._fitted_spectrum = fitted_spectrum
+
+        # Update component model parameters with fitted values
+        self._update_parameters_from_fit()
+
+        self.save_enabled = True
+
+    def vue_register_spectrum(self, event):
+        """
+        Add a spectrum to the data collection based on the currently displayed
+        parameters (these could be user input or fit values).
+        """
+        # Make sure the initialized models are updated with any user-specified
+        # parameters
+        self._update_initialized_parameters()
+
+        # Need to run the model fitter with run_fitter=False to get spectrum
+        model, spectrum = fit_model_to_spectrum(self._spectrum1d,
+                                                self._initialized_models.values(),
+                                                self.model_equation)
+
+        self.n_models += 1
+        label = self.model_label
+        if label in self.data_collection:
+            self.app.remove_data_from_viewer('spectrum-viewer', label)
+            # Some hacky code to remove the label from the data dropdown
+            temp_items = []
+            for data_item in self.app.state.data_items:
+                if data_item['name'] != label:
+                    temp_items.append(data_item)
+            self.app.state.data_items = temp_items
+            # Remove the actual Glue data object from the data_collection
+            self.data_collection.remove(self.data_collection[label])
+        self.data_collection[label] = spectrum
+        self.save_enabled = True
+        self.data_collection.add_link(
+            LinkSame(self.data_collection[self._label_to_link].pixel_component_ids[0],
+                     self.data_collection[label].pixel_component_ids[0]))
+
```

Comment:
You should be able to automatically display the added spectrum by making a call similar to below.

```suggestion
        self.data_collection.add_link(
            LinkSame(self.data_collection[self._label_to_link].pixel_component_ids[0],
                     self.data_collection[label].pixel_component_ids[0]))

    self.app.add_data_to_viewer('spectrum-viewer', data_label=label)
```

Reply:
I created a [follow-up issue](https://github.com/spacetelescope/jdaviz/issues/162) to figure out what's going on with the `add_data_to_viewer` call, with the intent that we'll leave it commented out for now and make the user manually check the model spectrum in the data dropdown to show it.